src/wallet/wallet.cpp

```diff
@@ -1748,6 +1748,21 @@ int64_t CWallet::RescanFromTime(int64_t startTime, const WalletRescanReserver& r
     return startTime;
 }
 
+GCSFilter::ElementSet CWallet::GetScriptPubKeysForBlockFilter()
+{
+    GCSFilter::ElementSet ret;
+    for (const auto& i : CBasicKeyStore::GetAllWatchedScriptPubKeys()) {
+        ret.emplace(i.begin(), i.end());
+    }
+    for (const auto& bs : CBasicKeyStore::GetAllBareScripts()) {
```

Comment:
Just because we know about a script does not imply we're watching it I believe.

Reply:
Sure, but I'd rather err on the safe side of scanning a block too much than missing a block (and thus lose coins)